custom-wasm-transformers/samples/wasm-java-helm-chart/add-custom-files-dirs/main.go

```diff
@@ -0,0 +1,272 @@
+func DirectoryDetect(dir string) (map[string][]Artifact, error) {
+	dataFilePath := filepath.Join(dir, PomFile)
+	if f, err := os.Stat(dataFilePath); err == nil && !f.IsDir() {
+		serviceName, err := getServiceName(dataFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get the service name. Error: %w", err)
+		}
+		services := map[string][]Artifact{}
+		services[serviceName] = []Artifact{{
+			Paths: map[string][]string{"ProjectPath": {dir}},
+		}}
+		return services, nil
+	}
+	return nil, nil
+}
+
+func Transform(
+	newArtifacts []Artifact,
+	alreadySeenArtifacts []Artifact,
+) ([]PathMapping, []Artifact, error) {
+	pathMappings := []PathMapping{}
+	artifacts := []Artifact{}
+	pathTemplate := "{{ SourceRel .ServiceFsPath }}"
+	for _, v := range newArtifacts {
+		_serviceName, ok := v.Configs["Service"].(map[string]interface{})
+		if !ok {
+			fmt.Println("failed to v.Configs[Service].(map[string]interface{})")
+			continue
+		}
+		// serviceName, ok := _serviceName["serviceName"].(string)
+		serviceName, ok := _serviceName["ServiceName"].(string)
+		if !ok {
+			fmt.Printf("failed to _serviceName[ServiceName].(string). actual _serviceName: %T %+v\n", _serviceName, _serviceName)
+			continue
+		}
+		if len(v.Paths) == 0 {
+			fmt.Printf("failed to v.Paths. actual v.Paths: %T %+v\n", v.Paths, v.Paths)
+			continue
+		}
+		projPaths := v.Paths["ProjectPath"]
+		if len(projPaths) == 0 {
+			fmt.Printf("failed to v.Paths[ProjectPath]. actual v.Paths: %T %+v\n", projPaths, projPaths)
+			continue
+		}
+		fmt.Printf("len(projPaths) %d %+v\n", len(projPaths), projPaths)
+		dir := projPaths[0]
+		// Create a path template for the service
+		pathTemplateName := strings.ReplaceAll(serviceName, "-", "") + "path"
+		tplPathData := map[string]interface{}{
+			"ServiceFsPath":    dir,
+			"PathTemplateName": pathTemplateName,
+		}
+		pathMappings = append(pathMappings,
+			PathMapping{
+				Type:           "PathTemplate",
+				SrcPath:        pathTemplate,
+				TemplateConfig: tplPathData,
+			})
+		// Since the helm chart uses the same templating character {{ }} as Golang templates,
+		// we use `SpecialTemplate` type here where the templating character is <~ ~>.
+		// The `Template` type can be used for all normal cases
+		pathMappings = append(pathMappings, PathMapping{
+			Type:     "SpecialTemplate",
+			DestPath: "{{ ." + pathTemplateName + " }}",
+			TemplateConfig: map[string]interface{}{
+				"ServiceFsPath": dir,
+				"ServiceName":   serviceName,
+			}})
+		pathMappings = append(pathMappings, PathMapping{
+			Type:     "Source",
+			SrcPath:  "{{ ." + pathTemplateName + " }}",
+			DestPath: "{{ ." + pathTemplateName + " }}",
+		})
+	}
+	return pathMappings, artifacts, nil
+}
+
+//go:export RunDirectoryDetect
+func RunDirectoryDetect(
+	inputJsonPtr uint32,
+	inputJsonLen uint32,
+	outputJsonPtr uint32,
+	outputJsonLen uint32,
+) int32 {
+	fmt.Println("mycustomtransformer: RunDirectoryDetect start")
```

Comment:
General comment: Let us use log package as much as possible.

Reply:
we don't use `log` package anywhere in our code, either in Move2Kube or in any of the related repos.